lib/ui/fixtures/ui_test.dart

```diff
@@ -1076,3 +1076,39 @@ external void _callHook(
   Object? arg20,
   Object? arg21,
 ]);
+
+Scene CreateRedBoxScene(Size size) {
+  final SceneBuilder builder = SceneBuilder();
+  builder.pushOffset(0.0, 0.0);
+  final Paint paint = Paint()
+    ..color = Color.fromARGB(255, 255, 0, 0)
+    ..style = PaintingStyle.fill;
+  final PictureRecorder baseRecorder = PictureRecorder();
+  final Canvas canvas = Canvas(baseRecorder);
+  canvas.drawRect(Rect.fromLTRB(0.0, 0.0, size.width, size.height), paint);
+  final Picture picture = baseRecorder.endRecording();
+  builder.addPicture(Offset(0.0, 0.0), picture);
+  builder.pop();
+  return builder.build();
+}
+
+@pragma('vm:entry-point')
+@pragma('vm:external-name', 'NotifyNative')
+external void notifyNative();
+
+@pragma('vm:entry-point')
+void incorrectImmediateRender() {
+  PlatformDispatcher.instance.views.first.render(CreateRedBoxScene(Size(2, 2)));
+}
+
+@pragma('vm:entry-point')
+void incorrectDoubleRender() {
+  PlatformDispatcher.instance.onBeginFrame = (Duration value) {
+    PlatformDispatcher.instance.views.first.render(CreateRedBoxScene(Size(2, 2)));
+    PlatformDispatcher.instance.views.first.render(CreateRedBoxScene(Size(3, 3)));
+  };
+  PlatformDispatcher.instance.onDrawFrame = () {
+    PlatformDispatcher.instance.views.first.render(CreateRedBoxScene(Size(4, 4)));
+    PlatformDispatcher.instance.views.first.render(CreateRedBoxScene(Size(5, 5)));
+  };
+}
```

Comment:
Please request a frame after registering the frame callbacks. `dart:ui` apps should not rely on the engine to schedule a frame.

Reply:
I don't think it works in this case. A requested frame goes to the engine and the animator, which don't exist in this test case.

Reply:
Ahh right, gotcha. Could add a quick comment that the test calls these callbacks "directly" without using the full frame lifecycle?